src/core/auth/authentication/components/KratosUI.tsx

```diff
@@ -56,65 +62,56 @@ const KratosMessages: FC<{ messages?: Array<UiText> }> = ({ messages }) => {
   );
 };
 
-const toUiControl = (node: UiNode, key: number) => {
-  const attributes = node.attributes;
-  if (isUiNodeInputAttributes(attributes)) {
-    const variant = guessVariant(node);
-
-    const extraProps: KratosInputExtraProps = {
-      autoCapitalize: 'off',
-      autoCorrect: 'off',
-    };
-
-    switch (variant) {
-      case 'email':
-      case 'username':
-        extraProps.autoComplete = 'username';
-        break;
-      case 'password':
-        extraProps.autoComplete = 'password';
-        break;
-    }
+interface SocialCustomization {
+  icon: FC<React.SVGProps<SVGSVGElement> & { title?: string }>;
+  theme: { palette: { primary: { main: string } } };
+  label: string;
+}
 
-    switch (attributes.type) {
-      case 'hidden':
-        return <KratosHidden key={key} node={node} />;
-      case 'submit':
-        return <KratosButton key={key} node={node} />;
-      case 'checkbox':
-        return <KratosCheckbox key={key} node={node} />;
-      default:
-        return <KratosInput key={key} node={node} {...extraProps} />;
-    }
-  } else {
-    return <KratosInput key={key} node={node} />;
-  }
+const socialCustomizations: Record<string, SocialCustomization> = {
+  linkedin: {
+    theme: linkedInTheme,
+    icon: LinkedInIcon,
+    label: 'linkedin',
+  },
+  microsoft: {
+    theme: microsoftTheme,
+    icon: MicrosoftIcon,
+    label: 'microsoft',
+  },
 };
 
-export const KratosUI: FC<KratosUIProps> = ({ resetPasswordComponent, flow, ...rest }) => {
-  const { t } = useTranslation();
-  const [showFormAlert, setShowFormAlert] = useState(false);
-
-  const handleSubmit = useCallback((e: FormEvent<HTMLFormElement>) => {
-    const button = getActiveElement() as any;
-    // do ckeck if only submitting password method
-    if (button && button.name === 'method' && button.value === 'password') {
-      if (!e.currentTarget.checkValidity()) {
-        setShowFormAlert(true);
-        e.preventDefault();
-        e.stopPropagation();
-        return;
-      }
-    }
-  }, []);
+interface NodeGroups {
+  default: UiNode[];
+  oidc: UiNode[];
+  password: UiNode[];
+  rest: UiNode[];
+  submit: UiNode[];
+  hidden: UiNode[];
+}
 
-  type NodeGroups = { default: UiNode[]; oidc: UiNode[]; password: UiNode[]; rest: UiNode[] };
+export const KratosUI: FC<KratosUIProps> = ({
+  ui,
+  resetPasswordElement,
+  acceptTermsComponent: AcceptTerms,
+  buttonComponent: Button = AuthActionButton,
+  renderAcceptTermsCheckbox = checkbox => <KratosAcceptTermsCheckbox node={checkbox} />,
+  children,
+  ...rest
+}) => {
+  const { t, i18n } = useTranslation();
 
-  const nodesByGroup = useMemo(() => {
-    if (!flow) return;
+  const kratosFormContext = useKratosFormContext();
 
-    return flow.ui.nodes.reduce(
+  const nodesByGroup = useMemo(() => {
+    return ui?.nodes.reduce(
       (acc, node) => {
+        if (node.group !== 'oidc' && node.attributes['type'] === 'submit') {
```

Comment:
this is magic code, you will forget what it means in 5 days, extract to a method with a meaningful name as this is in the core of the logic for building the component